### ISSUE TITLE
Defined component name, that breaches componentName, doesn't throw

### ISSUE THREAD

**Issue by dryoma** [postcss/postcss-bem-linter#84]
Hi!

The README says:
> If a component is defined and the component name does not match your `componentName` pattern, the plugin will throw an error.

Yet this code
```
/** @define my_-__Component */

.my_-__Component-block {
  display: inline;
}
```
passes fine with `{ preset: 'suit' }` or the config:
```
{
  componentName: /[A-Z][a-zA-Z0-9]+/,
  componentSelectors: function(componentName) {
    return new RegExp('\\.' + componentName + '(?:-[a-zA-Z]+)?');
  }
}
```
`my_-__Component` definitely doesn't pass the `componentName` pattern. Am I getting the line from the docs wrong or is it really a mistake?

**Comment by davidtheclark** [comment]
Isn't that a legit selector?

If your component name is X and you're using SUIT, the following should be fine:
- `X-descendent`
- `X--modifier`
- `X-descendent--modifier`
- etc.

Your `.my_-__Component-block` is equivalent to `X-descendent`, right?

**Comment by davidtheclark** [comment]
Looks to me like it should be complaining. Maybe a bug, will have to look into it.

**Comment by davidtheclark** [comment]
@dryoma: The problem was that the regexp pattern lacked `^` and `$` to indicate the beginning and end of the string, meaning that it was satisfied if it just found one letter. Proper pattern is: `/^[A-Z][a-zA-Z0-9]+$/`.

**Comment by dryoma** [comment]
Indeed, it solves the issue. But the version with fixed preset isn't available in npm yet, could this be because the version in package.json is still 2.4.0?

**Comment by davidtheclark** [comment]
Whoops, I must have forgotten to publish. Thanks for pointing that out.

**Comment by davidtheclark** [comment]
Published in 2.4.1

**Comment by dryoma** [comment]
Thanks!
